File: API/Entities/OrderAggregate/Order.cs
using System.ComponentModel.DataAnnotations;

namespace API.Entities.OrderAggregate
{
    public class Order
    {
        public int Id { get; set; }
        public string BuyerId { get; set; }

        [Required]
        public ShippingAddress ShippingAddress { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        public List<OrderItem> OrderItems { get; set; }
        public int TotalProducts { get; set; }
    }
}

File: client/src/app/models/order.ts
export interface ShippingAddress {
  fullName: string;
  descriptionAddress: string;
}

export interface OrderItem {
  productId: number;
  name: string;
  pictureUrl: string;
  cod: number;
  weight: number;
  quantity: number;
}

export interface Order {
  id: number;
  buyerId: string;
  shippingAddress: ShippingAddress;
  orderDate: string;
  orderItems: OrderItem[];
  totalProducts: number;
}
